xds/src/test/java/io/grpc/xds/ControlPlaneRule.java

```diff
@@ -76,15 +76,23 @@ public class ControlPlaneRule extends TestWatcher {
   private static final String EDS_NAME = "eds-service-0";
   private static final String SERVER_LISTENER_TEMPLATE_NO_REPLACEMENT =
       "grpc/server?udpa.resource.listening_address=";
-  private static final String SERVER_HOST_NAME = "test-server";
   private static final String HTTP_CONNECTION_MANAGER_TYPE_URL =
       "type.googleapis.com/envoy.extensions.filters.network.http_connection_manager.v3"
           + ".HttpConnectionManager";
 
+  private String serverHostName;
   private Server server;
   private XdsTestControlPlaneService controlPlaneService;
   private XdsNameResolverProvider nameResolverProvider;
 
+  public ControlPlaneRule(String serverHostName) {
```

Comment:
Constructor is fine, but FYI you can also expose a fluent setter. That's generally how rules are configured for optional stuff.